op-node/rollup/engine/engine_controller.go

```diff
@@ -75,9 +76,9 @@ type EngineController struct {
 	safeAttrs    *derive.AttributesWithParent
 }
 
-func NewEngineController(engine ExecEngine, log log.Logger, metrics derive.Metrics, rollupCfg *rollup.Config, syncMode sync.Mode) *EngineController {
+func NewEngineController(engine ExecEngine, log log.Logger, metrics derive.Metrics, rollupCfg *rollup.Config, syncCfg *sync.Config, engineKind EngineClientKind) *EngineController {
```

Comment:
@axelKingsley wondering if adding the engineKind to the syncConfig is actually a clean place for it to live? Currently, the receives of sync configs are also the ones interested in the engine kind.